hi-ml/src/health_ml/run_ml.py

```diff
@@ -94,8 +96,11 @@ def setup(self, azure_run_info: Optional[AzureRunInfo] = None) -> None:
                         raise ValueError(f"Invalid setup: The dataset at index {i} is None")
                     local_datasets.append(check_dataset_folder_exists(dataset))
                 self.container.local_datasets = local_datasets  # type: ignore
-        # Ensure that we use fixed seeds before initializing the PyTorch models
-        seed_everything(self.container.get_effective_random_seed())
+        # Ensure that we use fixed seeds before initializing the PyTorch models.
+        # MONAI needs a separate method to make all transforms deterministic by default
+        seed = self.container.get_effective_random_seed()
+        seed_monai_if_available(seed)
```

Comment:
for my own understanding - to make the transforms deterministic we need both to fix the determinism in MONAI and to  fix a seed in the transform?

Reply:
No, seeding MONAI separately is more of a preventative measure. If someone accidentally adds randomized transforms (and forgets to seed them), the code added here will probably be enough to still make it reproducible

Reply:
I like the extra caution :)